src/main/java/pages/LoginPage.java

```diff
@@ -216,5 +216,12 @@ public String getTryDemoCustomerAccessSectionBackgroundColorAsHex() {
     public String getTryDemoCustomerAccessSectionFontColorAsHex() {
         return getColorAsHex(driver, tryDemoCustomerAccessSection, CssProperty.COLOR);
     }
+
+    public LoginPage loginWithInvalidEmailAndPassword(String email, String password) {
+        fillInInputField(driver.findElement(emailInputBy), email);
+        fillInInputField(driver.findElement(passwordInputBy),password);
+        driver.findElement(signInButton).click();
```

Comment:
This three steps (filling email, password and clicking on sign in button) are the same as in loginWithEmailAndPassword.

The metod should be renamed to tryToLoginWithEmailAndPassword() and then reused in loginWithEmailAndPassword method.

`public LoginPage tryToLoginWithEmailAndPassword(String email, String password) {
        fillInInputField(driver.findElement(emailInputBy), email);
        fillInInputField(driver.findElement(passwordInputBy),password);
        driver.findElement(signInButton).click();
        return this;
    }`
    
    and then :
    `public Homepage loginWithEmailAndPassword(String email, String password) {
        tryToLoginWithEmailAndPassword(email, password);
        return new Homepage(driver);
    }`